eclipse-collections-api/src/main/java/org/eclipse/collections/api/map/MapIterable.java

```diff
@@ -251,6 +252,19 @@
     Pair<K, V> detect(Predicate2<? super K, ? super V> predicate);
 
     /**
+     * Return the first key and value of the map as an Optional for which the predicate evaluates to true when
+     * they are given as arguments. The predicate will only be evaluated until such pair is found or until all
+     * of the keys and values of the map have been used as arguments. That is, there may be keys and values of
+     * the map that are never used as arguments to the predicate.
+     * <p>
+     * <pre>e.g.
+     * peopleByCity.detectOptional((city, person)
+     *          -> city.getName().equals("Anytown") && person.getLastName().equals("Smith"));
+     * </pre>
+     */
+    Optional<Pair<K, V>> detectOptional(Predicate2<? super K, ? super V> predicate);
```

Comment:
It's an interesting point. I'd still prefer to be able to call detect*() methods on UnifiedSet for example. Do you think it doesn't make much sense to call detect*() methods on non-OrderedIterable or non-OrderedMap?